rpc-server.py

```diff
@@ -21,9 +24,31 @@
 cab.open(name, test_vault_path)
 
 
-def check_auth(username, password):
-    # return username == 'myname' and password == 'secret'
-    return True
+def generateToken(username, password, vault_path):
+    SECURITY_SALT = bcrypt.gensalt().decode("utf-8")
+    auth_payload = {
+        'uid': username,
+        'password': password,
+        'vault_path': vault_path
+    }
+    token = create_token(SECURITY_SALT, auth_payload)
+    session['username'] = username
+    session['vault_path'] = vault_path
+    session['token'] = token
+    session.permanent = True
+
+
+def check_auth(token):
+    token = request.headers.get('token')
+    if 'token' in session:
+        return token == session['token']
+    return False
+
+
+@jsonrpc.method('App.login(username=str, password=str, vault_path=str) -> str')
+def login(username, password, vault_path):
+    generateToken(username, password, vault_path)
```

Comment:
Yes, indeed, my idea for this PR isn't to add the login check, but to add support for authenticating the requests that comes after login, using a token, instead of using the username and password every time.